src/features/history/effects.ts

```diff
@@ -0,0 +1,235 @@
+            );
+            return false;
+        }
+
+        if (
+            newMaxSize < MINIMUM_MAX_NUMER_OF_LINES ||
+            newMaxSize > MAXIMUM_MAX_NUMBER_OF_LINES
+        ) {
+            logger.error(
+                `Cannot set max number of lines in history file to ${newMaxSize}, it must be between ${MINIMUM_MAX_NUMER_OF_LINES} and ${MAXIMUM_MAX_NUMBER_OF_LINES}.`
+            );
+            return false;
+        }
+        dispatch(setMaxNumberOfLinesInHistoryFile(newMaxSize));
+
+        const currentNumberOfLinesInHistoryFile =
+            historyBuffer.getNumberOfLines();
+
+        if (currentNumberOfLinesInHistoryFile > newMaxSize) {
+            logger.debug(
+                `New history max size ${newMaxSize} is less than the current history file.`
+            );
+            dispatch(trimHistoryFileToMaxSize);
+        }
+    };
+
+export const getHistoryPercentage: AppThunk<RootState> = (
+    _dispatch,
+    getState
+) => {
+    const historySize = getNumberOfLinesInHistory(getState());
+    const maximumHistorySize = getMaximumNumberOfLinesInHistory(getState());
+
+    return (historySize / maximumHistorySize) * 100;
+};
+
+export const writeHistoryLine =
+    (line: string): AppThunk<RootState> =>
+    (dispatch, getState) => {
+        if (historyBuffer == null) {
+            logger.error(
+                'History Buffer is not initialized, this means that the app is not working as it should, and you should try to restart it. If the issue persists, please create a ticket on https://devzone.nordicsemi.com/'
+            );
+            return;
+        }
+
+        historyBuffer.pushLineToHistory(line);
+
+        const maximumNumberOfLinesInHistory = getMaximumNumberOfLinesInHistory(
+            getState()
+        );
+
+        if (historyBuffer.getNumberOfLines() > maximumNumberOfLinesInHistory) {
+            historyBuffer.trimDownToNumberOfLinesToKeep(
+                maximumNumberOfLinesInHistory
+            );
+        }
+
+        dispatch(writeHistory(historyBuffer.getHistory()));
+    };
+
+export const trimHistoryFile =
+    (numberOfLinesToKeep: number): AppThunk<RootState> =>
+    (dispatch, getState) => {
+        if (historyBuffer == null) {
+            logger.error(
+                'History Buffer is not initialized, this means that the app is not working as it should, and you should try to restart it. If the issue persists, please create a ticket on https://devzone.nordicsemi.com/'
+            );
+            return;
+        }
+
+        const maximumNumberOfLines = getMaximumNumberOfLinesInHistory(
+            getState()
+        );
+
+        if (
+            numberOfLinesToKeep > maximumNumberOfLines ||
+            numberOfLinesToKeep < 0
+        ) {
+            logger.error(`Lines to keep can never be negative 🤪`);
```

Comment:
Not sure if the first condition of `numberOfLinesToKeep > maximumNumberOfLines` is actually possible but in that case, the error message doesn't relate to it.
Unless I misunderstand, which may very well be the case...